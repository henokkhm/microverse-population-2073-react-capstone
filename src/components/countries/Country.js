import PropTypes from 'prop-types';

import styles from '../../styles/Country.module.css';
import rightArrow from '../../assets/icons/circled-arrow.svg';

function Country({ name, populationIn2073 }) {
  return (
    <li className={styles.country}>
      <div className={styles.countryName}>{name}</div>
      <div>{`${populationIn2073.toLocaleString()} people`}</div>
      <button className={styles.arrowBtn} type="button">
        <img
          className={styles.arrowBtnIcon}
          src={rightArrow}
          alt="circled arrow button pointing right"
        />
      </button>
    </li>
  );
}

Country.propTypes = {
  name: PropTypes.string.isRequired,
  populationIn2073: PropTypes.number.isRequired,
};

export default Country;
