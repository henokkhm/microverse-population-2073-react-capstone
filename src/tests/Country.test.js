import { render, screen } from '@testing-library/react';

import Country from '../components/countries/Country';

describe('The Country component', () => {
  it('renders country name and population', () => {
    const countryData = {
      id: '6de33d15-f97e-4005-850f-8fc37f17a79c',
      name: 'Ethiopia [mock]',
      populationIn2073: 278863945,
    };

    render(
      <Country
        key={countryData.id}
        name={countryData.name}
        populationIn2073={countryData.populationIn2073}
      />,
    );

    expect(screen.getByText(/Ethiopia/i)).toBeInTheDocument();
    expect(screen.getByText(/278,863,945/i)).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const countryData = {
      id: '6de33d15-f97e-4005-850f-8fc37f17a79c',
      name: 'Ethiopia [mock]',
      populationIn2073: 278863945,
    };
    const { asFragment } = render(
      <Country
        key={countryData.id}
        name={countryData.name}
        populationIn2073={countryData.populationIn2073}
      />,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
