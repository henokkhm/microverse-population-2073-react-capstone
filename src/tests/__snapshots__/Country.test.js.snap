// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Country component matches snapshot 1`] = `
<DocumentFragment>
  <li
    class="country"
  >
    <div
      class="countryName"
    >
      Ethiopia [mock]
    </div>
    <div>
      278,863,945 people
    </div>
    <button
      class="arrowBtn"
      type="button"
    >
      <img
        alt="circled arrow button pointing right"
        class="arrowBtnIcon"
        src="circled-arrow.svg"
      />
    </button>
  </li>
</DocumentFragment>
`;
